lib/device.js

```diff
@@ -251,6 +256,169 @@ const util = require('util'),
+                        // it's directory path. Use it as destination dir Path
+                        captureFileName = _createDateFileName();
+                        destinationPath = path.resolve(options.outputPath);
+                    }
+                }
+
+                options.captureFormat = captureFormat;
+                options.captureFileName = captureFileName;
+                options.captureDirPath = "/tmp/capture",
+                options.sourcePath = path.join(options.captureDirPath, captureFileName);
+                options.destinationPath = destinationPath;
+                options.ignore = true;
+                options.silent = true;
+                
+                next();
+            }
+
+            function _createDateFileName() {
+                // Capture file name generate : targetname_displayX_yyyymmddhhmmss.png"
+                function __getDate(){
+                    const curDate = new Date();
+                    const dateFormat = __pad(curDate.getFullYear(), 2)
+                        +  __pad((curDate.getMonth()+1), 2)
+                        + __pad(curDate.getDate(), 2)
+                        + __pad(curDate.getHours(), 2)
+                        + __pad(curDate.getMinutes(), 2)
+                        + __pad(curDate.getSeconds(), 2);
+                    return dateFormat;
+                }
+
+                function __pad(number, length) {
+                    let str = '' + number;
+                    while (str.length < length) {
+                        str = '0' + str;
+                    }
+                    return str;
+                }
+
+                const fileName = options.session.target.name + "_" + "display" + options.display+ "_" + __getDate() + ".png";
+                return fileName;
+            }
+
+            function _captureFile(next) {
+                log.info("device#capture#_captureFile()");
+                const cmd = "/bin/mkdir -p " + options.captureDirPath;
+                options.session.run(cmd, null, null, null, function(err) {
+                    if (err) {
+                        return setImmediate(next, err);
+                    } else {
+                        options.targetCaptureDir = true;
+                        const target = options.session.getDevice(),
+                            addr = target.lunaAddr.captureCompositorOutput,
+                            param = {
+                                // luna param
+                                subscribe: false,
+                                output : options.sourcePath,
+                                format: options.captureFormat,
+                                displayId : options.display
+                            };
+
+                        luna.send(options, addr, param, function(lineObj, next) {
+                            log.silly("device#capture#_captureFile():", "lineObj:", lineObj);
+                            if (lineObj.returnValue) {
```

Comment:
we need a error handle when lineObj.returnValue is null or fail return or occured error.

Reply:
Please check luna.send() in luna.js.
 The function has error handle logic when lineObj.returnValue is null or false.